src/geode/mesh/core/tetrahedral_solid.cpp

```diff
@@ -325,6 +322,85 @@ namespace geode
         return result;
     }
 
+    template < index_t dimension >
+    absl::optional< PolyhedronFacet >
+        TetrahedralSolid< dimension >::polyhedron_adjacent_facet(
+            const PolyhedronFacet& polyhedron_facet ) const
+    {
+        const auto opt_polyhedron_adj =
+            this->polyhedron_adjacent( polyhedron_facet );
+        if( !opt_polyhedron_adj )
+        {
+            return absl::nullopt;
+        }
+        std::array< index_t, 3 > vertices;
+        for( const auto v : LRange{ 3 } )
+        {
+            vertices[v] =
+                this->polyhedron_facet_vertex( { polyhedron_facet, v } );
+        }
+        const auto polyhedron_adj = opt_polyhedron_adj.value();
+        const std::array< index_t, 4 > adj_vertices{
```

Comment:
Why not calling polyhedron_vertices()

Reply:
It leads to more Ir

Reply:
Yes, probably but you might want to benchmark time on this change only. Not sure if this is really relevant.